packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Writers/TypeProviderWriter.cs

```diff
@@ -9,153 +9,175 @@ namespace Microsoft.Generator.CSharp
     public class TypeProviderWriter
     {
         protected readonly TypeProvider _provider;
-        private readonly CodeWriter _writer;
 
         public TypeProviderWriter(TypeProvider provider)
-            : this(new CodeWriter(), provider)
         {
+            _provider = provider;
         }
 
-        private TypeProviderWriter(CodeWriter writer, TypeProvider provider)
+        public virtual CodeFile Write()
```

Comment:
Ok so this is the place why it is public.
I get it now - and I will revoke my above comment about the accessibility of this type